src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.AnalyzerExecutionContext.cs

```diff
@@ -212,18 +212,18 @@ void processMembers(IEnumerable<ISymbol> members)
             [Conditional("DEBUG")]
             private void VerifyNewEntryForPendingMemberSymbolsMap(ISymbol symbol, HashSet<ISymbol>? dependentSymbols)
             {
-                Debug.Assert(_lazyPendingMemberSymbolsMap != null);
+                Debug.Assert(_lazyPendingMemberSymbolsMap != null, $"{nameof(_lazyPendingMemberSymbolsMap)} was expected to be a non-null value.");
 
                 if (_lazyPendingMemberSymbolsMap.TryGetValue(symbol, out var existingDependentSymbols))
                 {
                     if (existingDependentSymbols == null)
                     {
-                        Debug.Assert(dependentSymbols == null);
+                        Debug.Assert(dependentSymbols == null, $"{nameof(dependentSymbols)} was expected to be null.");
                     }
                     else
                     {
-                        Debug.Assert(dependentSymbols != null);
-                        Debug.Assert(dependentSymbols.SetEquals(existingDependentSymbols));
+                        Debug.Assert(dependentSymbols != null, $"{nameof(dependentSymbols)} was expected to be a non-null value.");
+                        Debug.Assert(dependentSymbols.SetEquals(existingDependentSymbols), $"{nameof(dependentSymbols)} was expected to contain the same elements in {nameof(existingDependentSymbols)}.");
```

Comment:
Do I need to add more information to this message? Such as the elements in both sets? Or this will make it unnecessarily long?

Reply:
I think it is pretty likely that this is the failing assert. Can you change this code to use an if statement that checks for the set equality and on failure dumps the fully qualified names of all symbols in both sets? We will allocate these large strings only when the assert fails.

Reply:
@mavasani Thanks for review. I applied your suggestion.